lib/ClangImporter/ImportType.cpp

```diff
@@ -2726,9 +2726,14 @@ static ParamDecl *getParameterInfo(ClangImporter::Implementation *impl,
       !isa<clang::CXXConstructorDecl>(param->getDeclContext()) &&
       impl->isDefaultArgSafeToImport(param) &&
       !param->isTemplated()) {
-    SwiftDeclSynthesizer synthesizer(*impl);
-    if (CallExpr *defaultArgExpr = synthesizer.makeDefaultArgument(
-            param, swiftParamTy, paramInfo->getParameterNameLoc())) {
+    FuncDecl *defaultArgFunc;
+    CallExpr *defaultArgExpr;
+    std::tie(defaultArgFunc, defaultArgExpr) =
+        SwiftDeclSynthesizer::makeDefaultArgument(
+            param, swiftParamTy, paramInfo->getParameterNameLoc(),
+            impl->ImportedHeaderUnit);
+    if (defaultArgFunc && defaultArgExpr) {
```

Comment:
I think there is this invariant that either both of these are null or both of these are non-null. So I wonder if we should turn the second part of the condition into an assertion to catch cases when only one of them is non-null.